app/controllers/projects/members_controller.rb

```diff
@@ -110,25 +109,51 @@ def find_member(str)
       User.find_by(login: str.to_s.sub(/ <.*>$/, ""))
     end
 
-    # Redirects back to show_project.
-    def update_member_status(project, candidate)
-      admin = member = :remove
-      case params[:commit]
-      when :change_member_status_make_admin.l
-        unless project.is_admin?(@user)
+    def update_membership(project, candidate)
+      unless update_trust_status(project, candidate)
+        unless update_admin_status(project, candidate)
           return must_be_project_admin!(project.id)
         end
 
-        admin = member = :add
-      when :change_member_status_make_member.l
-        member = :add
+        update_member_status(project, candidate)
+      end
+      return_to_caller(project, params[:target])
+    end
+
+    def update_trust_status(project, candidate)
+      if params[:commit] == :change_member_status_revoke_trust.l
+        flash_notice(:change_member_status_revoke_trust_flash.l)
```

Comment:
Should any of these be `.t` rather than `.l`? (Both here and below.)
(Else why go to the trouble of having a translation string?)

Reply:
My understanding of this is fuzzy, but it's that `.l` "localize" is generally to be preferred unless the string is itself calling other embedded translation strings, and therefore needs the `.t`. I would appreciate any clarification!

Reply:
I used .l since the original code had .l.  As Nimmo points out .t is only needed in the case that there are embedded translation strings and in these case they are kind of being used as constants so .l seemed reasonable to me.  Don't have a strong opinion.